benchmark.h

```diff
@@ -322,12 +326,81 @@ class Benchmark {
                 << "," << searcher.name()
                 << std::endl;
     }
+    if (csv_) {
+      PrintResultCSV(index);
+    }
+  }
+
+  template<class Index>
+  void PrintResultCSV(const Index& index) {
+    const std::string filename = "./results/" + dataset_name_ + "_results_table.csv";
+
+    std::ofstream fout(filename, std::ofstream::out | std::ofstream::app);
+
+    if (!fout.is_open()) {
+      std::cerr << "Failure to print CSV on " << filename << std::endl;
+      return;
+    }
+
+    if (track_errors) {
+      fout << index.name()
+           << "," << index.variant()
+           << "," << log_sum_search_bound_
+           << "," << l1_sum_search_bound_
+           << "," << l2_sum_search_bound_
+           << std::endl;
+      return;
+    }
+
+    if (build) {
+      fout << index.name()
+           << "," << index.variant()
+           << "," << build_ns_
+           << "," << index.size()
+           << std::endl;
+      return;
+    }
+
+    if (cold_cache) {
+      double ns_per = ((double)individual_ns_sum) / ((double)lookups_.size());
```

Comment:
Where is `individual_ns_sum` set? And please avoid C-style casts. Use `static_cast` instead.

Reply:
This code's basically the same as the existing PrintResult function since I only adapted it to produce CSV files instead, so I wouldn't know why that decision was made. I will fix the casting in both functions.